### PR TITLE
[SPARK-37129][TESTS] Supplement Java 17 benchmark results created by GitHub Actions machines

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr adds all benchmark result files running in Java 17, the file generation uses the same way as SPARK-34950.

### Why are the changes needed?
To have a base line of the benchmarks accordingly.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
It was generated from:

- [Run benchmarks: * (JDK 17)](https://github.com/LuciferYang/spark/actions/runs/1390580580)

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
https://github.com/apache/spark/pull/34409 got merged. Let's fix it in here together.

**Review by srowen — APPROVED** [review]

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ on:
         required: true
         default: '*'
       jdk:
-        description: 'JDK version: 8 or 11'
+        description: 'JDK version: 8, 11 or 17'
```
[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply .github/workflows/benchmark.yml:30]: @LuciferYang just to clarify, are there more benchmarks to generate results? or are these all?

[the remark this answers is not in the document]
Comment (LuciferYang) [reply .github/workflows/benchmark.yml:30]: TPCDSQueryBenchmark-jdk17-results.txt is missing here, and the previous TPCDSQueryBenchmar kbenchmark results are not generated by GA

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by LuciferYang** [comment]
thanks all

**Pull request closed without merging**